addons/udes_stock_routing/models/stock_rule.py

```diff
@@ -36,7 +36,9 @@ def get_path_from_location(self, location):
             ]
         )
         if push_steps:
-            return push_steps.sorted(key=lambda p: p.location_src_id.parent_path.strip("/").split("/")[-1], reverse=True)[0]
+            to_order = push_steps.filtered(lambda p: p.location_src_id.parent_path)
```

Comment:
Cant we search where parent_path of the location_src_id has value in the search, rather then filtering after searched